web/api/backend.go

```diff
@@ -113,45 +114,62 @@ func Handler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		// If the structure of the body is wrong, return an HTTP error.
 		w.WriteHeader(http.StatusBadRequest)
+		io.WriteString(w, "Unable to Parse JSON")
 		return
 	}
 
-	// Checking if there is a token to use if the user asks to use a token or a credential body. Will return an error if those components are missing.
-	if (requestBody.UseToken && len(requestBody.Token) == 0) || (len(requestBody.Credential) == 0) {
-		w.WriteHeader(http.StatusBadRequest)
+	// JSON BODY representing the credentials attribute to be used (i.e. either the credentials found in the token or inputted).
+	creds := make(map[string]interface{})
+
+	// JSON Body representing the credentials attribute with the following format: "credentials":{"scopes":[],...}
+	// This is so to match with how the wrapper class will save the credentials file to memory.
+	credsString := make(map[string]interface{})
+
+	// If command type is test or token, credentials are not necessary
+	if !(reflect.DeepEqual(requestBody.CommandType, "test")) || (reflect.DeepEqual(requestBody.CommandType, "info")) || (reflect.DeepEqual(requestBody.CommandType, "reset")) {
```

Comment:
Nested/complicated if conditions can make the code a bit harder to read, how about extracting the logic into some helpers here:

```go
if (isCredentialsNeeded(requestBody)) {
 if (requestBody.UseToken) {
    // This getToken helper will check for edge cases (empty token) and verify the token
    creds, err  = getToken(requestBody)
    if (err != nil) {
      io.WriteString(w, err.message)
      return
    }
 } else {
     cred = requestBody.Credential
 }
 ...
}
```

Reply:
Added a function called isCredentialsNeeded to check the logic